apps/total-typescript/src/pages/api/progress/[lesson].ts

```diff
@@ -0,0 +1,60 @@
+import {withSentry} from '@sentry/nextjs'
+import {NextApiRequest, NextApiResponse} from 'next'
+import {getSdk} from '@skillrecordings/database'
+import z from 'zod'
+
+const Subscriber = z.object({
+  id: z.number(),
+  first_name: z.string().optional(),
+  email_address: z.string(),
+  state: z.string(),
+})
+
+type Subscriber = z.infer<typeof Subscriber>
+
+const lessonProgressHandler = async (
+  req: NextApiRequest,
+  res: NextApiResponse,
+) => {
+  if (req.method === 'POST') {
+    try {
+      const {findOrCreateUser, toggleLessonProgressForUser} = getSdk()
+      const lessonSlug = req.query.lesson as string
+      const subscriberCookie = req.cookies['ck_subscriber']
+
+      if (!subscriberCookie) {
+        res.status(200).end()
+        return
+      }
+
+      const subscriber = Subscriber.parse(JSON.parse(subscriberCookie))
+
+      if (!subscriber) {
+        res.status(200).end()
+        return
+      }
+
+      const {user} = await findOrCreateUser(subscriber.email_address)
```

Comment:
we use email as identity so we can just create a user based on subscriber's email address. ez